python/hsfs/constructor/query.py

```diff
@@ -148,12 +149,12 @@ def join(
                 the join. Defaults to `[]`.
             join_type: Type of join to perform, can be `"inner"`, `"outer"`, `"left"` or
                 `"right"`. Defaults to "inner".
-
+            prefix: User provided prefix to avoid feature name clash. Defaults to `None`.
```

Comment:
maybe explain that it will be applied to the "right" feature group that's being joined